### PR TITLE
centralize version declaration of jna, junit, protobuf

### PR DESCRIPTION
**Pull request by soloturn**
centralize version declaration of jna, junit, protobuf

### CODE REVIEW

File: engine-tests/build.gradle.kts [hunk 1]
```diff
@@ -53,7 +53,7 @@ dependencies {
     // Dependency not provided for modules, but required for module-tests
     implementation(libs.gson)
     implementation("org.codehaus.plexus:plexus-utils:3.0.16")
-    implementation("com.google.protobuf:protobuf-java:3.16.1")
+    implementation("com.google.protobuf:protobuf-java:${libs.versions.protobuf.get().toString()}")
```
Comment (BenjaminAmos) [inline engine-tests/build.gradle.kts:56]: For protobuf `implementation()` dependencies, is there a reason why you could not use `implementation(libs.protobuf.java)` or similar? As far as I understand the linked issue, you only need to use a string in the `protoc` block.

File: engine/build.gradle.kts [hunk 2]
```diff
@@ -152,7 +152,7 @@ dependencies {
 
 protobuf {
     protoc {
-        artifact = "com.google.protobuf:protoc:3.22.0"
+        artifact = "com.google.protobuf:protoc:${libs.versions.protobuf.get().toString()}"
```
Comment (BenjaminAmos) [inline engine/build.gradle.kts:156]:
```suggestion
        // See https://github.com/google/protobuf-gradle-plugin/issues/563
        artifact = "com.google.protobuf:protoc:${libs.versions.protobuf.get().toString()}"
```

File: engine/build.gradle.kts [hunk 3]
```diff
@@ -62,7 +62,7 @@ dependencies {
     api(libs.gson)
     api("net.sf.trove4j:trove4j:3.0.3")
     implementation("io.netty:netty-all:4.1.77.Final")
-    implementation("com.google.protobuf:protobuf-java:3.22.0")
+    implementation("com.google.protobuf:protobuf-java:${libs.versions.protobuf.get().toString()}")
```
Comment (BenjaminAmos) [inline engine/build.gradle.kts:65]: You should be able to use `libs.protobuf.java` here as well (once you define it in the version catalogue).

File: engine-tests/build.gradle.kts (hunk 1 above)
Comment (soloturn) [reply engine-tests/build.gradle.kts:56]: i tried this first, and it did not work, and i still fail to understand why.

Comment (soloturn) [reply engine-tests/build.gradle.kts:56]: i tried it, but i did not find a way so it worked. it worked for you somehow? i do not understand WHY it does not work though ...

**Review by BenjaminAmos — APPROVED** [review]

**Pull request merged**